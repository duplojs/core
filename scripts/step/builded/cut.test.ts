import { readFile } from "fs/promises";
import { CutStep } from "../cut";
import { BuildedCutStep } from "./cut";
import { Response } from "@scripts/response";
import { resolve } from "path";
import { zod } from "@scripts/index";
import { duploTest } from "@test/utils/duploTest";

it("BuildedCutStep", async() => {
	const cutFunction = () => ({});

	const step = new CutStep(cutFunction, ["test"], [new Response(100, "toto", zod.undefined())]);
	const buildedCutStep = new BuildedCutStep(duploTest, step);

	expect(buildedCutStep.step).toBe(step);

	expect(buildedCutStep.cutFunction).toBe(cutFunction);

	expect(buildedCutStep.drop).toStrictEqual(["test"]);

	expect(buildedCutStep.toString(2)).toBe(
		await readFile(resolve(import.meta.dirname, "__data__/cut.txt"), "utf-8"),
	);
});
