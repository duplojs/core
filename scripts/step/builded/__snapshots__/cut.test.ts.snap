// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BuildedCutStep 1`] = `
"
		
/* beforeCutStep(index: 2) */
/* end_block */


		result = this.steps[2].cutFunction(floor, request);

		
/* beforeTreatResultCutStep(index: 2) */
/* end_block */


		if(result instanceof this.Response){

				let temp = this.steps[2].responseZodSchema.safeParse(result);

				if(!temp.success){
					throw new this.ContractResponseError(temp.error, result);
				}
			
break Execution;
}

		
/* beforeIndexingResultCutStep(index: 2) */
/* end_block */


		floor.drop("test", result["test"]);

		
/* afterCutStep(index: 2) */
/* end_block */

		"
`;
